### ISSUE TITLE
IINA displays subtitles for all episodes of a TV show

### ISSUE THREAD

**Issue by low-batt** [iina/iina#3908]
**System and IINA version:**

- macOS 12.4
- IINA 1.3.0

**Expected behavior:**
For a specific TV show episode IINA searches Open Subtitles and displays subtitles for that particular episode and not subtitles for any episode of the show.

**Actual behavior:**
Once issue #3907 is fixed, for a video file named `Better.Call.Saul.S04E03.1080p.BluRay.x265-RARBG.mp4` using Open Subtitles IINA finds 100 subtitles, displaying subtitles for any episode of the TV show. Since the number of subtitles is capped at 100, the correct subtitle file may not be among the 100 shown. Searching [Open Subtitles](https://www.opensubtitles.org/en/en) by hand using the text `Better.Call.Saul.S04E03.1080p.BluRay.x265-RARBG`, Open Subtitles adjusts the search string to be `better call saul s04e03` and finds the correct subtitle files for that particular episode.

**Steps to reproduce:**
- Make a copy of any video long enough to have subtitles
- Rename the copy to be `Better.Call.Saul.S04E03.1080p.BluRay.x265-RARBG.mp4`
- Open the video file in a version of IINA with issue #3907 fixed
- Click on `Find Online Subtitles...` under the `Subtitles` menu
- IINA displays 100 subtitles for various episodes

- [x] MPV does not have this problem.

This is an IINA issue.

**How often does this happen?**
Everytime.

**Comment by low-batt** [comment]
### Analysis

The code in `OpenSubSubtitle.requestIMDB` uses the key `IDMovieIMDB` as seen here:
```swift
let IMDB = (bestGuess?["IDMovieIMDB"] as? String) ?? ""
```

For a TV show that identifies the show, not an episode of the show.

### Fixing

There are issues with merge conflicts in this area with the plugin system, so I will be holding off creating a pull request until the plugin system has been merged.

The `BestGuess` returned by Open Subtitles:
<details>
<summary><b>Open Subtitles Results</b></summary>

```xml
<name>BestGuess</name>
<value>
  <struct>
    <member>
      <name>IDMovieIMDB</name>
      <value>
        <int>3032476</int>
      </value>
    </member>
    <member>
      <name>MovieName</name>
      <value>
        <string>Better Call Saul</string>
      </value>
    </member>
    <member>
      <name>MovieYear</name>
      <value>
        <string>2015</string>
      </value>
    </member>
    <member>
      <name>MovieKind</name>
      <value>
        <string>tv series</string>
      </value>
    </member>
    <member>
      <name>Reason</name>
      <value>
        <string>Data Intersection</string>
      </value>
    </member>
    <member>
      <name>Episode</name>
      <value>
        <int>3</int>
      </value>
    </member>
    <member>
      <name>Season</name>
      <value>
        <int>4</int>
      </value>
    </member>
    <member>
      <name>IMDBEpisode</name>
      <value>
        <string>7074002</string>
      </value>
    </member>
  </struct>
</value>
```
</details>

Contains a `IMDBEpisode` key in addition to the `IDMovieIMDB` key. If the `IMDBEpisode` key is present IINA should prefer the value of that key as it identifies the specific TV episode.

The fix not hard. The following would need to be cleaned up to avoid the duplicate code, but something like this, which includes the fix for #3907:
<details>
<summary><b>Fixed Code</b></summary>

```swift
guard let bestGuess = try self.findPath(["data", filename, "BestGuess"], in: response) as? [String: Any] else {
  resolver.fulfill("")
  return
}
if let imdbEntry = bestGuess["IMDBEpisode"] {
  if let IMDB = imdbEntry as? Int {
    resolver.fulfill(String(IMDB))
    return
  }
  let IMDB = imdbEntry as? String ?? ""
  resolver.fulfill(IMDB)
}
guard let imdbEntry = bestGuess["IDMovieIMDB"] else {
  resolver.fulfill("")
  return
}
if let IMDB = imdbEntry as? Int {
  resolver.fulfill(String(IMDB))
  return
}
let IMDB = imdbEntry as? String ?? ""
resolver.fulfill(IMDB)
```
</details>

With that fix IINA displays:
![issue-3908-subtitles](https://user-images.githubusercontent.com/86170219/184274760-31b9e82c-368a-4a83-9573-88a250f155ef.jpg)

**Comment by low-batt** [comment]
Added label to indicate fix has been merged into develop. Re-opening since this is still a problem in IINA 1.3.0, the latest release. This way any users planning on reporting this problem will hopefully find this open issue and not enter a duplicate.

**Comment by low-batt** [comment]
Fixed in IINA 1.3.1.
